### DISCUSSION TITLE
(False positive) "App Hanging" errors on iOS

### DISCUSSION THREAD

**Discussion by Amenti** [getsentry/sentry-dotnet#3758] [General]
Hi,

in our .Net Avalonia app for we're getting hundreds of App Hanging exceptions on iOS only without any users complaining about problems since one of the newer Sentry versions. macOS, Windows and Android are fine.

I'm not able to reproduce this in the simulator or during debug. 

I tried to filter out App Hanging exceptions like this (since there seems to be some discussion that this could be [somewhat expected behavior](https://github.com/react-native-clipboard/clipboard/issues/212)):

```
          options.SetBeforeSend(sentryEvent=>
            {
                // see: https://github.com/getsentry/sentry-cocoa/discussions/2715
                if (OperatingSystem.IsIOS() && sentryEvent.Message?.Formatted?.Contains("App Hanging"))
                {
                    return null;
                }

                return sentryEvent;
            });
```

But it doesn't help at all. Is there something I'm obviously doing wrong here?

Thanks in advance.

**Comment by jamescrosswell** [comment]
@Amenti you could disable AppHangTracking entirly on iOS (which seems to be what you're trying to do in the `SetBeforeSend` event)... So something like this:
```csharp
#if IOS
    options.Native.EnableAppHangTracking = false;
#endif
```

**Reply by Amenti** [reply]
Thanks a lot. I didn't see this option yet.
